nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-shared/src/main/java/org/apache/nifi/kafka/shared/component/KafkaClientComponent.java

```diff
@@ -107,6 +107,35 @@ public interface KafkaClientComponent {
             )
             .build();
 
+    PropertyDescriptor AWS_PROFILE_NAME = new PropertyDescriptor.Builder()
+            .name("aws.profile.name")
+            .displayName("AWS Profile Name")
+            .description("The AWS Profile to consider when there are multiple profiles available.")
+            .dependsOn(
+                    SASL_MECHANISM,
+                    SaslMechanism.AWS_MSK_IAM
+            )
+            .required(false)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    PropertyDescriptor AWS_DEBUG_CREDS = new PropertyDescriptor.Builder()
+            .name("aws.debug.creds")
+            .displayName("Debug AWS Credentials")
+            .description("This property helps to debug which AWS credential is being exactly used. If this property is set to true "
+                    + "and `software.amazon.msk.auth.iam.internals.MSKCredentialProvider` logger is set to DEBUG, a log will be printed "
+                    + "including IAM Account, IAM user id and the ARN of the IAM Principal corresponding to the credential being used. "
+                    + "It is recommended to use this property only during debug since it makes an additional remote call.")
```

Comment:
@nandorsoma Thanks for adding this new feature!
I had just a look at the code and would like to add a note here: I would avoid adding "debug" properties to processors because they are related to troubleshooting, not to the business functionality.
Would not it be possible to make it configurable in some other way? (e.g. java system property documented on the Additional Details page)

Reply:
One potential option could be to pass the debug status for the associated Processor's ComponentLog, although that would require some additional wiring. Either way, I agree that this is not ideal as a Processor property.